File: subclass/product.kt
abstract class Product (
  private var id: String,
  private var name: String,
  private var price: String,
)

class ElectronicProduct (
  private var id: String,
  private var name: String,
  private var price: String,
  private var productionDate: String,
) : Product(id, name, price)

class ConsumableProduct (
  private var id: String,
  private var name: String,
  private var price: String,
  private var expirationDate: String,
) : Product(id, name, price)

File: relation/generalization-and-specialization/product.dart
abstract class Product {
  String _id;
  String _name;
  String _price;

  Product(this._id, this._name, this._price);
}

class ElectronicProduct extends Product {
  String _id;
  String _name;
  String _price;
  String _productionDate;

  ElectronicProduct(
    String this._id,
    String this._name,
    String this._price,
  ) : super(_id, _name, _price);
}

class ConsumableProduct  extends Product {
  String _id;
  String _name;
  String _price;
  String _expirationDate;

  ConsumableProduct(
    String this._id,
    String this._name,
    String this._price,
  ) : super(_id, _name, _price);
}
